src/articles/FindArticles.js

```diff
@@ -88,6 +84,12 @@ export default function NewArticles() {
       });
     }
     document.title = "Zeeguu";
+
+    if (runningInChromeDesktop() || runningInFirefoxDesktop()) {
```

Comment:
because of this situation with the delay I wonder whether it would make sense to initialize `hasExtension` to `undefined`. That way you don't risk that the extension popup shows up on a machine that's very fast in rendering or so. As long as `hasExtension` is undefined, it means we don't know. So maybe `checkExtensionInstalled` should be called after 100ms and after 500ms, or something that clearly is more, we can do another callback that sets `hasExtension` to `false` if it is stil undefined. Does you see what I mean?

Reply:
This was a bit more tricky than I expected. It seems that if I set it in the useEffect, it still considers it undefinied even if it was set by the first timeout. My solution was to allow to pass a timeoutId which contains this function to set the variable to false, and if the code is injected, it deletes that function, meaning it won't be set, otherwise the timeout will run and it will be set to false. What do you think?

Reply:
I will push the current state so you can see what I mean

Reply:
Could we move this whole thing into a hook? It would be great if the whole logic wouldbe localized. It really is orthogonal to the main functionality of the `FindArticles` page.

Reply:
Sure! I can look into making this a Hook, should it just return two variables isRunningChrome, isRunningFirefox or should it return a constant?

Reply:
It might not be so important anymore. I think I was more thinking of getting rid of the timeoutTimer - which you already did :) I guess if we really had a hook, it should probably only return `showExtensionReminder` as a boolean?